### PR TITLE
feat: import website from aestus upstream, solve conflicts

### PR DESCRIPTION
**Pull request by merklefruit**
* Added the `website` crate from upstream: https://github.com/aestus-relay/helix/tree/constraints-api-v0/crates/website/src
* Unfortunately I couldn't preserve the commit history across forks, but I can add co-authorship!
* Ran `just fmt fmt-check clippy` with the pinned toolchain.
* Set the default listen url to unspecified `0.0.0.0`
* Solved some minor clippy lints

### CODE REVIEW

File: Dockerfile
Comment (thedevbirb) [inline Dockerfile:1]: I'd restore some changes made in this Dockerfile as this is what Gattaca uses and works for them.

File: crates/cmd/Cargo.toml [hunk 1]
```diff
@@ -19,4 +20,5 @@ tracing.workspace = true
 tracing-subscriber.workspace = true
 tracing-appender.workspace = true
 
-num_cpus = "1.16.0"
\ No newline at end of file
+num_cpus = "1.16.0"
+futures = "0.3.28"
```
Comment (thedevbirb) [inline crates/cmd/Cargo.toml]: We can take this from the workspace `Cargo.toml` file

File: crates/cmd/src/main.rs [hunk 2]
```diff
@@ -56,7 +59,45 @@ async fn run() {
...
+        handles.push(website_handle);
+    }
+
+    if !handles.is_empty() {
+        // Await all tasks to prevent the runtime from shutting down
+        futures::future::join_all(handles).await;
+    } else {
+        tracing::error!("No services are enabled.");
+        std::process::exit(1);
+    }
```
Comment (thedevbirb) [inline crates/cmd/src/main.rs]: Can we clean up this logic a bit? Feels a bit cluttered for being the `main` function. Also the semantics of `enabled_routes` might not match with what's done here. Can you double check it?

File: crates/database/Cargo.toml [hunk 3]
```diff
@@ -40,3 +40,9 @@ rand = "0.8"
 dashmap.workspace = true
 thiserror.workspace = true
 tracing.workspace = true
+tokio-postgres-rustls.workspace = true
+rustls.workspace = true
+rustls-native-certs.workspace = true
+rustls-pemfile = "2.2.0"
+rustls-pki-types = "1.10.0"
+
```
Comment (thedevbirb) [inline crates/database/Cargo.toml]: These can be inherited from the workspace as well. Also.. are they needed?

File: rust-toolchain.toml [hunk 4]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.72.0"
+channel = "1.77.0"
```
Comment (thedevbirb) [inline rust-toolchain.toml]: Is it truly necessary to bump the version for this PR? I'm not against it, just checking if it's needed or not.

File: crates/database/src/postgres/postgres_db_service.rs
Comment (thedevbirb) [inline crates/database/src/postgres/postgres_db_service.rs:1]: Do we need all the certificate logic in order to display the website? I don't think so.

**Review by thedevbirb — CHANGES_REQUESTED** [review]
Some comments. I'd re-check what are the minimum requirements to serve the website and remove everything unnecessary and move them to separate PRs if needed.

File: crates/cmd/src/main.rs (hunk 2 above)
Comment (merklefruit) [reply crates/cmd/src/main.rs]: Reverted to the original (almost)

File: crates/database/Cargo.toml (hunk 3 above)
Comment (merklefruit) [reply crates/database/Cargo.toml]: no, rm'd

File: rust-toolchain.toml (hunk 4 above)
Comment (merklefruit) [reply rust-toolchain.toml]: restored

File: crates/database/src/postgres/postgres_db_service.rs
Comment (merklefruit) [reply crates/database/src/postgres/postgres_db_service.rs:1]: rm'd

**Comment by merklefruit** [comment]
I tried to minimize the diffs, now testing this version on our devnet.

Edit: seems OK.

This branch has been pushed as image `ghcr.io/chainbound/helix:v0.3.0-alpha.rc4`

**Review by thedevbirb — APPROVED** [review]

**Pull request merged**